File: codeforces/Div3_966/d.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <set>
#include <map>
#include <string>
#define ll long long
#define ull unsigned long long

using namespace std;

vector<ll> tab(2e5 + 2);

void    solution()
{
    ll  n;
    cin >> n;
    vector<ll> v(n);
    ll i = 0, res = 0;
    while (i < n)
    {
        cin >>v[i];
        if (i == 0)
            tab[i] = v[i];
        else
            tab[i] = tab[i - 1] + v[i];
        i++;
    }
    string s;
    cin >> s;
    ll j =  n - 1;
    i = 0;
    while (1)
    {
        while (s[i] && s[i] != 'L')
            i++;
        while (j >= 0 && s[j] != 'R')
            j--;
        if (i >= j)
            break ;
        if (s[i] == 'L' && s[j] == 'R')
        {
            if (i - 1 >= 0)
                res += tab[j] - tab[i - 1];
            else
                res += tab[j];
            s[i] = '.';
            s[j] = '.';
        }
        i++;
        j--;
    }
    cout << res << endl;
}

int main()
{
    int t;
    cin >> t;
    while (t--)
        solution();
    return (0);
}

File: codeforces/Div3_966/d.java
import java.util.Scanner;

public class Main {

    static long[] tab = new long[200002];

    public static void solution() {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        long[] v = new long[n];

        int i = 0;
        long res = 0;

        while (i < n) {
            v[i] = sc.nextLong();
            if (i == 0) {
                tab[i] = v[i];
            } else {
                tab[i] = tab[i - 1] + v[i];
            }
            i++;
        }

        String s = sc.next();

        int pos1 = s.indexOf("L");
        int pos2 = s.lastIndexOf("R");

        while (pos1 != -1 && pos2 != -1) {
            if (pos2 > pos1) {
                if (pos1 - 1 >= 0) {
                    res += tab[pos2] - tab[pos1 - 1];
                } else {
                    res += tab[pos2];
                }
            }
            s = s.substring(0, pos1) + '.' + s.substring(pos1 + 1);
            s = s.substring(0, pos2) + '.' + s.substring(pos2 + 1);
            pos1 = s.indexOf("L");
            pos2 = s.lastIndexOf("R");
        }

        System.out.println(res);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            solution();
        }
    }
}
